provision/rosa-cross-dc/Taskfile.yaml

```diff
@@ -308,3 +307,185 @@ tasks:
+      MONITORING_UIDS:
+        sh: "cat {{.ROOT_DIR}}/.task/monitoring-uids-{{.ROSA_CLUSTER_NAME}}"
+      GRAFANA_PASSWORD:
+        sh: "aws secretsmanager get-secret-value --region eu-central-1 --secret-id keycloak-master-password --query SecretString --output text --no-cli-pager || echo admin"
+      GRAFANA_DOMAIN:
+        sh: "cat {{.ISPN_DIR}}/.task/kubecfg/ocp-prefix-{{.ROSA_CLUSTER_NAME}}"
+    cmds:
+      - >
+        KUBECONFIG="{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}"
+        helm -n monitoring upgrade --install grafana grafana/grafana --version {{.GRAFANA_VERSION}} -f "{{.KC_DIR}}/grafana.yaml"
+        --set ingress.hosts[0]="grafana.{{.GRAFANA_DOMAIN}}"
+        --set securityContext.runAsUser="{{.MONITORING_UIDS}}"
+        --set securityContext.runAsGroup="{{.MONITORING_UIDS}}"
+        --set securityContext.fsGroup="{{.MONITORING_UIDS}}"
+        --set adminPassword="{{.GRAFANA_PASSWORD}}"
+    sources:
+      - "{{.ROOT_DIR}}/.task/monitoring-uids-{{.ROSA_CLUSTER_NAME}}"
+      - "{{.ISPN_DIR}}/.task/kubecfg/ocp-prefix-{{.ROSA_CLUSTER_NAME}}"
+      - "{{.KC_DIR}}/grafana.yaml"
+      - "{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}"
+
+  install-jaeger:
+    label: "install-jaeger-{{.ROSA_CLUSTER_NAME}}"
+    internal: true
+    requires:
+      vars:
+        - ROSA_CLUSTER_NAME
+    vars:
+      RETENTION: '{{.RETENTION | default "168h"}}'
+      JAEGER_VERSION: '{{.JAEGER_VERSION | default "0.58.0"}}'
+    env:
+      KB_RETENTION: '{{.RETENTION}}'
+    cmds:
+      - >
+        KUBECONFIG="{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}"
+        helm upgrade --install jaeger jaegertracing/jaeger --version {{.JAEGER_VERSION}} -n monitoring -f "{{.ROOT_DIR}}/../minikube/jaeger/values.yaml"
+      - envsubst < "{{.ROOT_DIR}}/../minikube/jaeger/deployment-patch.yaml" > .task/jaeger-{{.ROSA_CLUSTER_NAME}}-patchfile.yaml
+      - >
+        KUBECONFIG="{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}"
+        oc patch deployment jaeger -n monitoring --patch-file .task/jaeger-{{.ROSA_CLUSTER_NAME}}-patchfile.yaml
+    sources:
+      - "{{.ROOT_DIR}}/../minikube/jaeger/values.yaml"
+      - "{{.ROOT_DIR}}/../minikube/jaeger/deployment-patch.yaml"
+      - "{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}"
+
+  install-grafana-charts:
+    label: "install-grafana-charts-{{.ROSA_CLUSTER_NAME}}"
+    internal: true
+    requires:
+      vars:
+        - ROSA_CLUSTER_NAME
+    cmds:
+      - >
+        KUBECONFIG="{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}"
+        helm upgrade --install monitoring "{{.KC_DIR}}/monitoring"
+    sources:
+      - "{{.KC_DIR}}/monitoring/*"
+      - "{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}"
+
+
+  monitoring:
+    desc: "Install grafana dashboards in both ROSA clusters"
+    deps:
+      - common:split
+      - common:env
+    requires:
+      vars:
+        - ROSA_CLUSTER_NAME_1
+        - ROSA_CLUSTER_NAME_2
+    cmds:
+      - task: install-helms
```

Comment:
AFAIK the `helm` commands also target a specific cluster, so running this without a parameter would target only the currently logged in cluster.

Reply:
Those are local helm commands to add the Grafana and Jaeger repositories. I changed the task name to make it more clear.

Reply:
I didn't know that those command were local only, so I learned something new today!